packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -82,25 +94,59 @@ contract BridgePool is Testable {
         bytes32 indexed priceRequestAncillaryDataHash,
         address indexed caller
     );
-    event ProvidedLiquidity(address indexed token, uint256 amount, uint256 lpTokensMinted, address liquidityProvider);
+
+    event LiquidityAdded(address indexed token, uint256 amount, uint256 lpTokensMinted, address liquidityProvider);
+    event LiquidityRemoved(address indexed token, uint256 amount, uint256 lpTokensBurnt, address liquidityProvider);
 
     modifier onlyFromOptimisticOracle() {
         require(msg.sender == address(_getOptimisticOracle()), "Caller must be OptimisticOracle");
         _;
     }
 
-    constructor(address _bridgeAdmin, address _timer) Testable(_timer) {
+    // TODO: should we consider changing the name of the LP token as a function of the l1Token? if so, might not be able
+    // to do this with this contract inheriting from expanded ERC20 or might need this contract to have an instance
+    // of the LPToken.
+    constructor(
+        address _bridgeAdmin,
+        address _l1Token,
+        address _timer
+    ) Testable(_timer) ExpandedERC20("UMA Insured Bride LP Token", "UMA-LP", 18) {
         bridgeAdmin = BridgeAdminInterface(_bridgeAdmin);
         require(bridgeAdmin.finder() != address(0), "Invalid bridge pool factory");
+
+        l1Token = IERC20(_l1Token);
     }
 
     /*************************************************
      *          LIQUIDITY PROVIDER FUNCTIONS         *
      *************************************************/
 
-    function deposit(address l1Token, uint256 amount) public {}
+    function addLiquidity(uint256 l1TokenAmount) public {
```

Comment:
+1 on having this be in terms of L1 tokens, not LP tokens that you want to create!